src/ut/mementoappserver_test.cpp

```diff
@@ -515,6 +515,10 @@ TEST_F(MementoAppServerWithDialogIDTest, OnInDialogRequestTest)
   pjsip_msg* rsp = parse_msg(msg.get_response());
   as_tsx.on_response(rsp, 0);
 
+  // On a non-BYE in dialog request, nothing is written to the store
+  EXPECT_CALL(*_helper, send_request(_)).WillOnce(Return(0));
+  as_tsx.on_in_dialog_request(parse_msg(msg.get_request()));
```

Comment:
You should really test this in a separate `MementoAppServerTsx` (you can re-use the old `_helper`) and then send in the `200 OK` to test that we don't hit the database incorrectly there.

Can you also do the BYE in a new transaction?